```
 Fail job if Python cache restore failed0s
##[error]Process completed with exit code 1.
Run echo "Failed to restore Python virtual environment from cache"
Failed to restore Python virtual environment from cache
##[error]Process completed with exit code 1.
```

--- a/homeassistant/components/recorder/util.py
+++ b/homeassistant/components/recorder/util.py
@@ -168,7 +168,7 @@ def validate_sqlite_database(dbpath: str) -> bool:
 
 
 def run_checks_on_open_db(dbpath, cursor):
-    """Run checks that will generate a databsae exception if there is corruption."""
+    """Run checks that will generate a sqlite3 exception if there is corruption."""
     if basic_sanity_check(cursor) and last_run_was_recently_clean(cursor):
         _LOGGER.debug(
             "The quick_check will be skipped as the system was restarted cleanly and passed the basic sanity check"

--- a/homeassistant/components/recorder/util.py
+++ b/homeassistant/components/recorder/util.py
@@ -158,22 +158,27 @@ def validate_sqlite_database(dbpath: str) -> bool:
 
     try:
         conn = sqlite3.connect(dbpath)
-        cursor = conn.cursor()
-        if last_run_was_recently_clean(cursor) and basic_sanity_check(cursor):
-            _LOGGER.debug(
-                "The quick_check will be skipped as the system was restarted cleanly and passed the basic sanity check"
-            )
-        else:
-            _LOGGER.debug(
-                "A quick_check is being performed on the sqlite3 database at %s", dbpath
-            )
-            cursor.execute("PRAGMA QUICK_CHECK")
+        run_checks_on_open_db(dbpath, conn.cursor())
         conn.close()
     except sqlite3.DatabaseError:
         _LOGGER.exception("The database at %s is corrupt or malformed.", dbpath)
         return False
 
     return True
+
+
+def run_checks_on_open_db(dbpath, cursor):
+    """Run checks that will generate a databsae exception if there is corruption."""
+    if basic_sanity_check(cursor) and last_run_was_recently_clean(cursor):
+        _LOGGER.debug(
+            "The quick_check will be skipped as the system was restarted cleanly and passed the basic sanity check"
+        )
+        return
+
+    _LOGGER.debug(
+        "A quick_check is being performed on the sqlite3 database at %s", dbpath
+    )
+    cursor.execute("PRAGMA QUICK_CHECK")
 
 
 def _move_away_broken_database(dbfile: str) -> None:

--- a/tests/components/recorder/test_util.py
+++ b/tests/components/recorder/test_util.py
@@ -81,9 +81,13 @@ def test_validate_or_move_away_sqlite_database(hass, tmpdir, caplog):
 
     _corrupt_db_file(test_db_file)
 
+    assert util.validate_sqlite_database(dburl) is False
+
     assert util.validate_or_move_away_sqlite_database(dburl) is False
 
     assert "corrupt or malformed" in caplog.text
+
+    assert util.validate_sqlite_database(dburl) is False
 
     assert util.validate_or_move_away_sqlite_database(dburl) is True
 
@@ -121,6 +125,31 @@ def test_basic_sanity_check(hass_recorder):
 
     with pytest.raises(sqlite3.DatabaseError):
         util.basic_sanity_check(cursor)
+
+
+def test_combined_checks(hass_recorder):
+    """Run Checks on the open database."""
+    hass = hass_recorder()
+
+    cursor = hass.data[DATA_INSTANCE].engine.raw_connection().cursor()
+
+    assert util.run_checks_on_open_db("fake_db_path", cursor) is None
+
+    # We are patching recorder.util here in order
+    # to avoid creating the full database on disk
+    with patch("homeassistant.components.recorder.util.last_run_was_recently_clean"):
+        assert util.run_checks_on_open_db("fake_db_path", cursor) is None
+
+    with patch(
+        "homeassistant.components.recorder.util.last_run_was_recently_clean",
+        side_effect=sqlite3.DatabaseError,
+    ), pytest.raises(sqlite3.DatabaseError):
+        util.run_checks_on_open_db("fake_db_path", cursor)
+
+    cursor.execute("DROP TABLE events;")
+
+    with pytest.raises(sqlite3.DatabaseError):
+        util.run_checks_on_open_db("fake_db_path", cursor)
 
 
 def _corrupt_db_file(test_db_file):

